Applications/Utils/FileConverter/TecPlotTools.cpp

```diff
@@ -0,0 +1,410 @@
+        MeshLib::UseIntensityAs::DATAVECTOR,
+        vec_names[0]));
+    MeshLib::Properties& properties = mesh->getProperties();
+    for (std::size_t i=1; i<vec_names.size(); ++i)
+    {
+        auto* const prop = properties.createNewPropertyVector<double>(
+            vec_names[i], MeshLib::MeshItemType::Cell, 1);
+        if (!prop)
+        {
+            ERR("Error creating array \"%s\".", vec_names[i].c_str());
+            return -5;
+        }
+        prop->reserve(scalars[i].size());
+        std::copy(scalars[i].cbegin(), scalars[i].cend(), std::back_inserter(*prop));
+    }
+
+    std::size_t const delim_pos(file_name.find_last_of("."));
+    std::string const base_name(file_name.substr(0, delim_pos + 1));
+    std::string const extension(file_name.substr(delim_pos, std::string::npos));
+
+    INFO("Writing section #%i", write_count);
+    MeshLib::IO::VtuInterface vtu(mesh.get());
+    vtu.writeToFile(base_name + std::to_string(write_count++) + extension);
+    return 0;
+}
+
+/// If a geometry-section is encountered, it is currently ignored
+void skipGeometrySection(std::ifstream& in, std::string& line)
+{
+    while (getline(in, line))
+    {
+        if ((line.find("TITLE") != std::string::npos) ||
+            (line.find("VARIABLES") != std::string::npos) ||
+            (line.find("ZONE") != std::string::npos))
+            return;
+    }
+}
+
+/// Splits a TecPlot file containing multiple sections/zones into seperate files
+int splitFile(std::ifstream& in, std::string file_name)
+{
+    std::ofstream out;
+    std::string line("");
+    std::string name;
+    std::pair<std::size_t, std::size_t> dims(0,0);
+    std::size_t val_count(0), val_total(0);
+    std::size_t write_count(0);
+    while (getline(in, line))
+    {
+        if (line.find("TITLE") != std::string::npos)
+        {
+            if (dataCountError(out, name, val_count, val_total)) return -3;
+            writeTecPlotSection(out, file_name, write_count, val_count, val_total);
+            out << line << "\n";
+            continue;
+        }
+        else if (line.find("VARIABLES") != std::string::npos)
+        {
+            if (dataCountError(out, name, val_count, val_total)) return -3;
+            writeTecPlotSection(out, file_name, write_count, val_count, val_total);
+            out << line << "\n";
+            continue;
+        }
+        else if (line.find("ZONE") != std::string::npos)
+        {
+            if (dataCountError(out, name, val_count, val_total)) return -3;
+            writeTecPlotSection(out, file_name, write_count, val_count, val_total);
+            out << line << "\n";
+            name = getName(line);
+            dims = getDimensions(line);
+            val_total = dims.first * dims.second;
+            val_count = 0;
+            continue;
+        }
+
+        out << line << "\n";
+        val_count++;
+    }
+    if (dataCountError(out, name, val_count, val_total))
+        return -3;
```

Comment:
Why `-3`? It seems the method could return a bool  value.

Reply:
Different values indicate different reasons for failing. ```-3``` is indicating an index error.

Reply:
I put an explanation on the exit codes in the documentation. Independent of that, a detailed error-message is output in each case.